internal/opslevel/opslevel.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one-file package.
+
+// Package opslevel is a package meant to help utilize the OpsLevel API in GitHub actions.
+package opslevel
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/getoutreach/actions/internal/slack"
+	"github.com/opslevel/opslevel-go/v2022"
+	"github.com/pkg/errors"
+)
+
+// NewClient returns a new opslevel client configured with the token we expect to exist in
+// the environment.
+func NewClient() (*opslevel.Client, error) {
+	opslevelToken := strings.TrimSpace(os.Getenv("OPSLEVEL_TOKEN"))
+	if opslevelToken == "" {
+		return nil, errors.New("OPSLEVEL_TOKEN environment variable is empty")
+	}
+	return opslevel.NewGQLClient(opslevel.SetAPIToken(opslevelToken)), nil
+}
+
+// LifecycleToLevel maps lifecycle index to level index
+// we want to keep this at the index level in case names or other attributes change
+var LifecycleToLevel = map[int]int{
+	// In Development >= Beginner
+	0: 0,
+	// Private Beta >= Silver
+	1: 2,
+	// Public Beta >= Silver
+	2: 2,
+	// Public/GA >= Silver
+	3: 2,
+	// Ops >= Silver
+	4: 2,
+	// End-of-life >= Beginner
+	5: 0,
+}
+
+// Service is wrapper around the OpsLevel client that allows caching
+// and efficient interaction with an OpsLevel service
+type Service struct {
+	client  *opslevel.Client
+	service *opslevel.Service
+
+	// Used for caching retrieved values
+	repo           *opslevel.Repository
+	maturityReport *opslevel.MaturityReport
+}
+
+// NewService creates a new OpsLevelService by looking it up using an alias
+func NewService(client *opslevel.Client, alias string) (Service, error) {
+	service, err := client.GetServiceWithAlias("devbase")
+	if err != nil {
+		return Service{}, errors.Wrap(err, fmt.Sprintf("gettings details for %s", "devbase"))
```

Comment:
Am I missing something with the Sprintf?
```suggestion
		return Service{}, errors.Wrap(err, gettings details for devbase")
```